### DISCUSSION TITLE
[WARN]  Ti.selectedBackgroundGradient DEPRECATED in 10.0.0, in favor of Ti.backgroundSelectedGradient

### DISCUSSION THREAD

**Discussion by caspahouzer** [appcelerator/titanium_mobile#12969] [Q&A] [answered]
COMMUNITY QUESTION
Hi, i'm updating an iOS app with SDK 10.0.0.GA
I get this warning [WARN]  Ti.selectedBackgroundGradient DEPRECATED in 10.0.0, in favor of Ti.backgroundSelectedGradient , but i don't use this property anywhere (tested also with grep command).

I found what causes the warning (on a createTableViewRow()). If i use either of these 2 properties (backgroundImage / backgroundSelectedImage) the warning is displayed. If both are disabled, the warning is not displayed. I would say there is some problem.

**Comment by ewanharris** [comment, marked as answer]
It looks like this is coming from the below line in the configureBackground method of TableViewRowProxy. @garymathews, should the check be whether `if (!IS_NULL_OR_NIL([self valueForKey:@"selectedBackgroundGradient"])) {`?

https://github.com/appcelerator/titanium_mobile/blob/ed0fbbb7a595c64dbcf79f4138433a17116a0dd9/iphone/Classes/TiUITableViewRowProxy.m#L410-L412

Test code is 

```js
const win = Ti.UI.createWindow();
const row = Ti.UI.createTableViewRow({ backgroundImage: 'doesnt-need-to-exist.png' });
win.add(Ti.UI.createTableView({ data: [ row ] }));
win.open();

```

**Reply by nicolomonili** [reply]
Hi @ewanharris, i tried your modification, it seems to work.

**Comment by ULSashido** [comment]
This fix should be merged in the next Ti SDK

**Reply by ewanharris** [reply]
This was merged into the master branch, but was not deemed critical enough for 10.1.1 so will be released in 10.2.0
